plugins/zenoh-plugin-storage-manager/src/memory_backend/mod.rs

```diff
@@ -138,6 +138,8 @@ impl Storage for MemoryStorage {
 
     async fn on_sample(&mut self, mut sample: Sample) -> ZResult<StorageInsertionResult> {
         trace!("on_sample for {}", sample.key_expr);
+        // NOTE: this will lead to inconsistencies, with different timestamps being generated for same sample
```

Comment:
Our current approach fixes this issue by paying for extra data messages.

Assume we have a sample with no timestamp, and two replicas receiving it. The replicas will assign a timestamp locally. When they reconcile with the alignment protocol, the one with the latest timestamp wins, i.e., we will have only one sample left in the system anyway. But there will be some data exchanges to reconcile them. 

Do you think it is better to drop the message with an error, or pay the price of this extra data exchange? I prefer the second option since there is no sample lost.

Reply:
My concern is that with the more publication the storage gets without timestamp, the more traffic between storages is required for alignment. This might lead to system overload and instabilities that might be hard to analyse.

In my opinion it's anyway a bad design to not have publication timestamped at the source when those publications needs to be stored in replicated storages. And better have to forbid it.
So I would prefer we clearly state in doc that for storage replication usage it's required to activate timestamping in Publisher's Sessions. And that we log an error in replicas receiving a non-timestamped sample. When acknowledgement on PUT will be implemented, the replicas will return an error to the Publisher.

@Mallets do you agree ?

Reply:
After a discussion with @Mallets and @JEnoch , the conclusion is to force the user to timestamp the samples if they need replication. 
A follow up question would be whether we should force user to timestamp all entries that are meant to be stored. This would be required for our consolidation policies (in the future) to work correctly. 
For now, in this PR, I will enforce it only for replication.